diff --git a/src/core/dbus-unit.c b/src/core/dbus-unit.c
index 1a037b7..f0550e8 100644
--- a/src/core/dbus-unit.c
+++ b/src/core/dbus-unit.c
@@ -1300,6 +1300,8 @@ static int append_cgroup(sd_bus_message *reply, const char *p, Set *pids) {
                  * readable in the subtree proper. */

                 r = cg_read_pidref(f, &pidref);
+                if (r == -EIO)
+                        continue;
                 if (IN_SET(r, 0, -EOPNOTSUPP))
                         break;
                 if (r < 0udo
